### PR TITLE
Add private provider 2

### PR DESCRIPTION
**Pull request by laurapacilio**
Users can now add private providers to the Terraform Cloud private registry. At this time, private providers are not available in Terraform Cloud. Once a provider, provider version, and one or more provider platforms have been added through the API, users can view the provider in the Terraform Cloud UI and begin using it in configurations. 

To support this new feature launch, this PR makes the following documentation updates:
- Adds a new Publishing Private Providers page that will help users understand the process and requirements for adding a private provider. It will point users to the API endpoints (documented in the API page) that they need to manage providers. We don't duplicate the documentation on the API page.
- Updates the other private and public registry documentation accordingly. 
- Renames the publishing modules page (from publish.mdx --> publish-modules.mdx) for clarity and adds a redirect. 

------

I have:

- [x] Built the site locally or verified that the website preview successfully deployed. 
- [x] Verified that images and formatting render correctly for each changed page.
- [x] Added redirects for moved or deleted pages.
- [x] Updated the relevant sidebar navigation files for added, deleted, reordered, or renamed pages.
- [x] Reviewed this content for adherence to [Engineering](https://github.com/hashicorp/engineering-docs/blob/main/writing/style-guide.md) and [Learn](https://github.com/hashicorp/learn/blob/master/STYLE_GUIDE.md) style guides, especially for:
    - [x] Images are 2048 px wide, annotation color and format, images have alt text for accessibility
    - [x] UI elements are bolded
    - [x] Code is properly formatted
- [x] Performed a final copyedit to catch typos, punctuation, and other minor inconsistencies.
- [x] Discussed this content with engineering or tagged an engineering reviewer for technical accuracy.
- [x] Added one or more labels that describe both the type of change (e.g. clarification) and associated product (e.g. tfc). 

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201758754579181

### CODE REVIEW

**Review by madaleines — APPROVED** [review]
🎉 LGTM 🎉

File: content/cloud-docs/api-docs/private-registry/providers.mdx [hunk 1]
```diff
@@ -50,19 +50,21 @@ For publicly curated providers, the Terraform Cloud Registry acts as a proxy to
 - The public registry discovery endpoints have the path prefix provided in the [discovery document](/registry/api-docs#service-discovery) which is currently `/api/registry/public/v1`.
 - [Authentication](/cloud-docs/api-docs/#authentication) is handled the same as all other Terraform Cloud endpoints.
 
-## List public Terraform Registry Providers for an Organization
+## List Public Terraform Registry Providers for an Organization
```
Comment (bethanyr) [inline content/cloud-docs/api-docs/private-registry/providers.mdx:53]:
```suggestion
## List Terraform Registry Providers for an Organization
```

File: content/cloud-docs/api-docs/private-registry/providers.mdx [hunk 2]
```diff
@@ -50,19 +50,21 @@ For publicly curated providers, the Terraform Cloud Registry acts as a proxy to
...
 `GET /organizations/:organization_name/registry-providers`
 
+### Parameters
+
 | Parameter            | Description                                                    |
 | -------------------- | -------------------------------------------------------------- |
 | `:organization_name` | The name of the organization to list available providers from. |
 
-Lists the providers from the public Terraform Registry that included in the private registry for the specified organization.
+Lists the public providers included in the private registry for the specified organization.
```
Comment (bethanyr) [inline content/cloud-docs/api-docs/private-registry/providers.mdx:63]:
```suggestion
Lists the providers included in the private registry for the specified organization.
```

File: content/cloud-docs/api-docs/private-registry/gpg-keys.mdx [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+
+`POST /api/registry/:registry_name/v2/gpg-keys`
+
+### Parameters
+
+| Parameter            | Description          |
+| -------------------- | -------------------- |
+| `:registry_name`     | Must be `private`.   |
+
+Uploads a GPG Key to a private registry scoped with a namespace. The response will provide a "key-id", which is required to [Create a Provider Version](#create-a-provider-version) (see below).
```
Comment (madaleines) [inline content/cloud-docs/api-docs/private-registry/gpg-keys.mdx:55]: Need to update path and can also remove `(see below)`

Comment (madaleines) [reply content/cloud-docs/api-docs/private-registry/gpg-keys.mdx:55]:
```suggestion
Uploads a GPG Key to a private registry scoped with a namespace. The response will provide a "key-id", which is required to [Create a Provider Version](/cloud-docs/api-docs-private-registry/provider-versions-and-platforms#create-a-provider-version-private-provider) (see below).
```

File: content/cloud-docs/registry/publish-providers.mdx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+You must create a provider, at least one version, and at least one platform for that version before consumers can use the provider in configurations. Before you begin, confirm that you can access the provider binaries, public GPG signing key, `SHA256SUMS` file, and `SHA256SUMS.sig` file from at least one release. You will upload these artifacts in the following steps.
+
+To publish a new provider through the Terraform Cloud API:
+
+1. Use the [Add a GPG key endpoint](/cloud-docs/api-docs/providers#add-a-gpg-key-private-provider) to add the public key that matches the signing key for the release. The response contains a `key-id` that you will use to create a provider version.
+
+    ```json
+    "key-id": "32966F3FB5AC1129"
+    ```
+1. Use the [Create a Provider Version endpoint](/cloud-docs/api-docs/providers#create-a-provider-version-private-provider) to create a version with the `key-id` from the previous step. The response includes URL links that you will use to upload the `SHA256SUMS` and `SHA256.sig` files.
```
Comment (madaleines) [inline content/cloud-docs/registry/publish-providers.mdx:44]:
```suggestion
1. Use the [Create a Provider Version endpoint](/cloud-docs/api-docs-private-registry/provider-versions-and-platforms#create-a-provider-version-private-provider) to create a version with the `key-id` from the previous step. The response includes URL links that you will use to upload the `SHA256SUMS` and `SHA256.sig` files.
```

File: content/cloud-docs/registry/publish-providers.mdx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    ```
+
+1. Upload the `SHA256SUMS` and `SHA256SUMS.sig` files to the URLs returned in the previous step. The example command below uploads the files from a local machine.
+
+    ```
+    $ curl -T local-example/terraform-provider-random_3.1.3_SHA256SUMS
+      https://archivist.terraform.io/v1/object/dmF1b64hd73ghd63
+    ```
+
+1. Use the [Create a Provider Platform endpoint](/cloud-docs/api-docs/providers#create-a-provider-platform-private-provider) to create a platform for the version. Platforms are binaries that allow the provider to run on a particular operating system and architecture combination (e.g., Linux and AMD64). The response includes a `provider-binary-upload` URL that you will use to upload the binary file for the platform.
```
Comment (madaleines) [inline content/cloud-docs/registry/publish-providers.mdx]:
```suggestion
1. Use the [Create a Provider Platform endpoint](//cloud-docs/api-docs-private-registry/provider-versions-and-platforms#create-a-provider-platform-private-provider) to create a platform for the version. Platforms are binaries that allow the provider to run on a particular operating system and architecture combination (e.g., Linux and AMD64). The response includes a `provider-binary-upload` URL that you will use to upload the binary file for the platform.
```

File: content/cloud-docs/api-docs/private-registry/gpg-keys.mdx (hunk 3 above)
Comment (laurapacilio) [reply content/cloud-docs/api-docs/private-registry/gpg-keys.mdx:55]: done!!

File: content/cloud-docs/registry/publish-providers.mdx (hunk 4 above)
Comment (laurapacilio) [reply content/cloud-docs/registry/publish-providers.mdx]: done!

File: content/cloud-docs/registry/publish-providers.mdx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+GPG signing is required for private providers, and you must upload the public key of the GPG keypair used to sign the release. Refer to [Preparing and Adding a Signing Key](/registry/providers/publishing#preparing-and-adding-a-signing-key) for more details. Unlike the public Terraform Registry, the private registry does not automatically upload new releases. You must manually add new provider versions and the associated release files.
+
+## Publishing a Provider and Creating a Version
+
+You must create a provider, at least one version, and at least one platform for that version before consumers can use the provider in configurations. Before you begin, confirm that you can access the provider binaries, public GPG signing key, `SHA256SUMS` file, and `SHA256SUMS.sig` file from at least one release. You will upload these artifacts in the following steps.
+
+To publish a new provider through the Terraform Cloud API:
+
+1. Use the [Add a GPG key endpoint](/cloud-docs/api-docs/providers#add-a-gpg-key-private-provider) to add the public key that matches the signing key for the release. The response contains a `key-id` that you will use to create a provider version.
```
Comment (madaleines) [inline content/cloud-docs/registry/publish-providers.mdx]:
```suggestion
1. Use the [Add a GPG key endpoint](/cloud-docs/api-docs/api-docs-private-registry/gpg-keys#add-a-gpg-key-private-provider) to add the public key that matches the signing key for the release. The response contains a `key-id` that you will use to create a provider version.
```

Comment (laurapacilio) [reply content/cloud-docs/registry/publish-providers.mdx]: done!

**Pull request merged**